### DISCUSSION TITLE
How to add GiST indexes in a migration?

### DISCUSSION THREAD

**Discussion by kohenkatz** [clickbar/laravel-magellan#35] [Q&A] [answered]
Using Laravel's built-in types, I am able to do this in a migration:

```php
    public function up()
    {
        Schema::create('timezones', function (Blueprint $table) {
            $table->id('gid');
            $table->string('tzid', 80);
            $table->multiPolygon('geom');
        });

        DB::statement('CREATE INDEX timezones_geom ON timezones USING GIST (geom);');
    }
```

I switched to this library because Laravel's `multiPolygon` always creates a `GEOGRAPHY` type but I want a `GEOMETRY` type:

```php
    public function up()
    {
        Schema::create('timezones', function (Blueprint $table) {
            $table->id('gid');
            $table->string('tzid', 80);
            $table->magellanMultiPolygon('geom');
        });

        DB::statement('CREATE INDEX timezones_geom ON timezones USING GIST (geom);');
    }
```

This gives me an error when I run the migrations:

```
 2023_03_29_161151_create_timezones_table .............................................................................................. 111ms FAIL

   Illuminate\Database\QueryException 

  SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "timezones" does not exist (SQL: CREATE INDEX timezones_geom ON timezones USING GIST (geom);)
```

Is this because `createAddColumnClosure` defers the creation of the column, or is there something else going on? How can I add the index after the column is added?

**Comment by kohenkatz** [comment]
I see that `$table->spatialIndex('geom');` works, but why doesn't the `DB::statement` function?

**Reply by djfhe** [reply]
The error suggests, that the table `timezones` does not exists, when u are running the statement. This has nothing to do with the columns in the table. I just tested your second migration and it worked. Can you retry running the migration?

**Comment by kohenkatz** [comment, marked as answer]
It turns out this problem occurs if a project has BOTH magellan and mstaack/laravel-postgis installed. Uninstalling either one of them makes the migration work properly.
